### PR TITLE
Fix: Synchronize phpcs commands run on Travis and in composer scripts

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] synchronizes the `phpcs` commands run on Travis and in `composer` scripts
* [x] moves configuration of rules into `phpcs.xml`
* [x] runs `composer fix`
* [x] manually fixes issues `phpcbf` is unable to fix

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,9 +17,9 @@ before_script:
 script:
  - composer test-ci
  - vendor/bin/phpcs --config-set ignore_warnings_on_exit 1
- - vendor/bin/phpcs --standard=psr2 src
- - if [[ ${TRAVIS_PHP_VERSION:0:3} == "7.0" ]]; then vendor/bin/phpcs --standard=phpcs.xml src --ignore=tests/Sniffs; fi
- - if [[ ${TRAVIS_PHP_VERSION:0:3} == "7.0" ]]; then vendor/bin/phpcs --standard=codor.xml src -spn; fi
```
Comment (localheinz) [inline .travis.yml]: None of these condition will evaluate to `true`, but `phpcs` is run using these rules in the `test` script in [`composer.json`](https://github.com/bmitch/churn-php/blob/master/composer.json#L58-L61).

File: src/Configuration/Config.php [hunk 2]
```diff
@@ -115,27 +121,12 @@ public function getFileExtensions(): array
     {
         return $this->configuration['fileExtensions'] ?? self::FILE_EXTENSIONS_TO_PARSE;
     }
-
-    /**
-     * @param array $configuration The array containing the configuration values.
-     * @return void
-     */
-    private function validateConfigurationValues(array $configuration)
```
Comment (localheinz) [inline src/Configuration/Config.php]: Class was too long, inlining the method was the only way around the 200 lines limit.

File: src/Managers/FileManager.php [hunk 3]
```diff
@@ -61,20 +61,30 @@ public function getPhpFiles(array $paths): FileCollection
      * @param  string $path Path in which to look for .php files.
      * @return void
      */
-    private function getPhpFilesFromPath(string $path)
+    private function getPhpFilesFromPath(string $path): void
     {
         $directoryIterator = new RecursiveDirectoryIterator($path);
         foreach (new RecursiveIteratorIterator($directoryIterator) as $file) {
-            if (! in_array($file->getExtension(), $this->fileExtensions)) {
-                continue;
```
Comment (localheinz) [inline src/Managers/FileManager.php]: Too many indentation levels.

File: src/Managers/FileManager.php [hunk 4]
```diff
@@ -86,9 +96,7 @@ private function fileShouldBeIgnored(SplFileInfo $file): bool
     {
         foreach ($this->filesToIgnore as $fileToIgnore) {
             $regex = $this->patternToRegex($fileToIgnore);
-            if (preg_match("#{$regex}#", $file->getPathName())) {
-                return true;
-            }
+            return preg_match("#{$regex}#", $file->getPathName()) === 1;
```
Comment (localheinz) [inline src/Managers/FileManager.php]: Too many indentation levels.

File: src/Managers/ProcessManager.php [hunk 5]
```diff
@@ -75,10 +76,22 @@ private function getProcessResults(int $numberOfParallelJobs)
         }
 
         foreach ($this->runningProcesses as $file => $process) {
-            if ($process->isSuccessful()) {
-                $this->runningProcesses->forget($process->getKey());
```
Comment (localheinz) [inline src/Managers/ProcessManager.php]: Too many indentation levels.

File: src/Managers/FileManager.php [hunk 6]
```diff
@@ -84,14 +94,15 @@ private function getPhpFilesFromPath(string $path)
      */
     private function fileShouldBeIgnored(SplFileInfo $file): bool
     {
-        foreach ($this->filesToIgnore as $fileToIgnore) {
+        $path = $file->getPathname();
+
+        $matchingFilesToIgnore = array_filter($this->filesToIgnore, function (string $fileToIgnore) use ($path): bool {
             $regex = $this->patternToRegex($fileToIgnore);
-            if (preg_match("#{$regex}#", $file->getPathName())) {
-                return true;
```
Comment (localheinz) [inline src/Managers/FileManager.php]: Too many indentation levels.

Unfortunately I haven't found a better solution, let me know if you can think of one.

File: src/Processes/ChurnProcess.php [hunk 7]
```diff
@@ -43,7 +43,7 @@ public function __construct(File $file, Process $process, string $type)
      * Start the process.
      * @return void
      */
-    public function start()
+    public function start(): void
```
Comment (localheinz) [inline src/Processes/ChurnProcess.php]: All of the changes to method signatures of `public`, as well as `protected` methods of non-`final` classes should be considered as backwards-incompatible changes, I believe.

**Comment by Nyholm** [comment]
I think this PR looks alright but I’m too ignorant with phpcs to give it a proper review. I usually use prettyCI and then never worry about anything like this.

**Comment by localheinz** [comment]
@Nyholm 

Ha, looks like I was referencing the wrong configuration file - after modification the amount of changes required to make the build pass has significantly dropped!

**Comment by bmitch** [comment]
Thanks for this. I'll review as soon as I have some time to sit down and give it a good look over.

**Pull request closed without merging**